pkg/runtime.v2/framework/plugins/torch/torch.go

```diff
@@ -43,11 +46,87 @@ func (t *Torch) Name() string {
 	return Name
 }
 
-func (t *Torch) EnforceMLPolicy(info *runtime.Info) error {
+// TODO (andreyvelich): Add support for PyTorch elastic when JobSet supports Elastic Jobs.
+func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *kubeflowv2.TrainJob) error {
 	if info == nil || info.MLPolicy == nil || info.MLPolicy.Torch == nil {
 		return nil
 	}
-	// TODO: Need to implement main logic.
+
+	// TrainJob contains the actual information for the Trainer.
+	numNodes := info.MLPolicy.NumNodes
+	if trainJob.Spec.Trainer != nil && trainJob.Spec.Trainer.NumNodes != nil {
+		numNodes = trainJob.Spec.Trainer.NumNodes
+	}
+	info.Trainer.NumNodes = numNodes
+
+	numProcPerNode := info.MLPolicy.Torch.NumProcPerNode
+	if trainJob.Spec.Trainer != nil && trainJob.Spec.Trainer.NumProcPerNode != nil {
+		numProcPerNode = trainJob.Spec.Trainer.NumProcPerNode
+	}
+
+	// Update envs for Info object.
+	// Add PyTorch distributed "PET_" values for torchrun
+	// TODO (andreyvelich): Add validation to check that TrainJob doesn't have "PET_" envs.
+	infoEnvs := []corev1.EnvVar{
+		{
+			Name:  constants.TorchEnvNumNodes,
+			Value: fmt.Sprintf("%d", *numNodes),
```

Comment:
```suggestion
			Value: fmt.Sprintf("%d", ptr.Deref(numNodes, 1)),
```

Consindering null pointer should be better.